### PR TITLE
Updates to enable live testing in sovereign clouds for multiple services

### PR DESCRIPTION
**Pull request by Luyunmt**
We are running live Tests against other clouds like US Gov and Azure China Cloud. The goal is to check whether new azure sdk package work with other clouds or not.

Using this RR to trace and fix issues we encounter in the tests. Will close it once we finish all the tests and send PR per services.

Updated list:
Text analytics
Core
Extensions
Event hubs
Identity
key vault
Search
Storage account
Form recognizer
Service bus
Event grid
Tables

### CODE REVIEW

**Review by jsquire — APPROVED** [review]
The Event Hubs and Service Bus bits look good to me;  I have some concerns with China being very, very slow due to the management operations, but I'll monitor and adjust timeouts as needed based on the results.

File: sdk/core/Azure.Core.TestFramework/src/TestEnvironment.cs [hunk 1]
```diff
@@ -131,7 +131,7 @@ static TestEnvironment()
         /// <summary>
         ///   The URL of the Azure Authority host to be used for authentication. Recorded.
         /// </summary>
-        public string AuthorityHostUrl => GetRecordedOptionalVariable("AZURE_AUTHORITY_HOST");
+        public string AuthorityHostUrl => GetRecordedOptionalVariable("AZURE_AUTHORITY_HOST") ?? "https://login.microsoftonline.com/";
```
[the remark this answers is not in the document]
Comment (pakrym) [reply sdk/core/Azure.Core.TestFramework/src/TestEnvironment.cs:134]: Is the AZURE_AUTHORITY_HOST value produced by the New-TestResources? Why do we ever need to fall back?

File: sdk/storage/Azure.Storage.Files.Shares/tests/DirectoryClientTests.cs [hunk 2]
```diff
@@ -1050,7 +1050,8 @@ public async Task GetFileClient_SpecialCharacters(string directoryName, string f
             ShareFileClient fileFromDirectoryClient = InstrumentClient(directoryClient.GetFileClient(fileName));
             Response<ShareFileInfo> createResponse = await fileFromDirectoryClient.CreateAsync(Constants.KB);
 
-            Uri expectedUri = new Uri($"https://{TestConfigDefault.AccountName}.file.core.windows.net/{test.Share.Name}/{Uri.EscapeDataString(directoryName)}/{Uri.EscapeDataString(fileName)}");
+            string endpointSuffix = Environment.GetEnvironmentVariable("STORAGE_ENDPOINT_SUFFIX") ?? Constants.ConnectionStrings.DefaultEndpointSuffix;
```
Comment (pakrym) [inline sdk/storage/Azure.Storage.Files.Shares/tests/DirectoryClientTests.cs:1053]: I don't think we should use env vars directly here. cc @kasobol-msft

Comment (kasobol-msft) [reply sdk/storage/Azure.Storage.Files.Shares/tests/DirectoryClientTests.cs:1053]: https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Common/tests/Shared/TestConfigurationsTemplate.xml has service endpoints. these should be used instead. And looks like that template needs work to accommodate these other clouds.

**Pull request closed without merging**